internal/literals/split.go

```diff
@@ -0,0 +1,198 @@
+package literals
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	mathrand "math/rand"
+	ah "mvdan.cc/garble/internal/asthelper"
+)
+
+const maxChunkSize = 4
+const minCaseCount = 3
+
+type split struct{}
+
+// check that the obfuscator interface is implemented
+var _ obfuscator = split{}
+
+func splitIntoRandomChunks(data []byte) [][]byte {
+	if len(data) == 1 {
+		return [][]byte{data}
+	}
+
+	var chunks [][]byte
+	for {
+		if len(data) == 0 {
+			break
+		}
+		chunkSize := 1 + mathrand.Intn(maxChunkSize)
```

Comment:
I don't understand what you mean. Can you explain that?

Reply:
If you always add 1 to the chunkSize, it can be maxChunksize+1, which I find misleading.

You probably want `1+mathrand.Intn(maxChunkSize-1)`.

And increase the `maxChunkSize` constant by one.

Reply:
If you always add 1 to the chunkSize, it can be maxChunksize+1, which I find misleading.

You probably want `1+mathrand.Intn(maxChunkSize-1)`.

And increase the `maxChunkSize` constant by one.

Reply:
`mathrand.Intn` generates `[0,maxChunkSize]`, while I need `[1, maxChunkSize]`, for that I add 1.
If `1+mathrand.Intn(maxChunkSize-1)` is used, the range of values will be `[1, maxChunkSize-1)`, which will be even more confusing.

Reply:
You currently generate a value in this range
`[1, maxChunkSize+1]`.

If `1+mathrand.Intn(maxChunkSize-1)` is used, the range of values will be `[1, maxChunkSize)`.

Reply:
https://golang.org/pkg/math/rand/#Intn
Intn returns, as an int, a non-negative pseudo-random number in **[0,n)** from the default Source. 

Documentation says different

Reply:
And after the random number generation you add 1 to the generated number, effectively generating a number in [1, maxChunkSize+1).

Reply:
```patch
Index: internal/literals/split.go
<+>UTF-8
===================================================================
 const (
-	maxChunkSize = 4
+	maxChunkSize = 5
 	minCaseCount = 3
 )
 
@@ -26,7 +26,7 @@
 
 	var chunks [][]byte
 	for len(data) > 0 {
-		chunkSize := 1 + mathrand.Intn(maxChunkSize)
+		chunkSize := 1 + mathrand.Intn(maxChunkSize-1)
 		if chunkSize > len(data) {
 			chunkSize = len(data)
 		}

```

@lu4p, are you talking about it?

Reply:
Yes

Reply:
fixed

Reply:
@pagran Sorry I get it now the range is exclusive of the top number, will revert this soon.